lib/selinux.c

```diff
@@ -98,6 +98,136 @@ int reset_selinux_file_context (void)
 	return 0;
 }
 
+/*
+ *  libselinux uses this callback when access gets denied or other
+ *  events happen. If audit is turned on, messages will be reported
+ *  using audit netlink, otherwise they will be logged using the usual
+ *  channels.
+ *
+ *  Code adopted from systemd, which adopted from dbus.
+ */
+__attribute__((__format__ (printf, 2, 3)))
+static int log_callback (int type, const char *fmt, ...) {
+	va_list ap;
+	char *buf;
+	int r;
+#if HAVE_AUDIT
+	int fd;
+#endif
+
+	va_start (ap, fmt);
+	r = vasprintf (&buf, fmt, ap);
+	va_end (ap);
+
+	if (r < 0) {
+		free (buf);
+		return 0;
+	}
+
+	fprintf (stderr, _("%s: libselinux: %s\n"),
+			    Prog, buf);
+	SYSLOG ((LOG_WARN, "libselinux: %s", buf));
```

Comment:
Do we want to always do both (log + audit) or only fall back to SYSLOG if audit not enabled?

Reply:
switched to try `audit`, use `syslog` only as fallback